app/worker.php

```diff
@@ -72,56 +113,145 @@
+    return new Certificate($queue);
+}, ['queue']);
+Server::setResource('usage', function ($register) {
+    return new Stats($register->get('statsd'));
+}, ['register']);
+Server::setResource('queueForMigrations', function (Connection $queue) {
+    return new Migration($queue);
+}, ['queue']);
+Server::setResource('logger', function (Registry $register) {
+    return $register->get('logger');
+}, ['register']);
 
-Server::setResource('queueForFunctions', function (Registry $register) {
-    $pools = $register->get('pools');
-    return new Func(
-        $pools
-            ->get('queue')
-            ->pop()
-            ->getResource()
-    );
+Server::setResource('pools', function (Registry $register) {
+    return $register->get('pools');
 }, ['register']);
 
 Server::setResource('log', fn() => new Log());
 
-Server::setResource('logger', function ($register) {
-    return $register->get('logger');
-}, ['register']);
+/**
+ * Get Functions Storage Device
+ * @param string $projectId of the project
+ * @return Device
+ */
+Server::setResource('getFunctionsDevice', function () {
+    return function (string $projectId) {
+        return getDevice(APP_STORAGE_FUNCTIONS . '/app-' . $projectId);
+    };
+});
 
-Server::setResource('statsd', function ($register) {
-    return $register->get('statsd');
-}, ['register']);
+/**
+ * Get Files Storage Device
+ * @param string $projectId of the project
+ * @return Device
+ */
+Server::setResource('getFilesDevice', function () {
+    return function (string $projectId) {
+        return getDevice(APP_STORAGE_UPLOADS . '/app-' . $projectId);
+    };
+});
 
-Server::setResource('pools', function ($register) {
-    return $register->get('pools');
-}, ['register']);
+/**
+ * Get Builds Storage Device
+ * @param string $projectId of the project
+ * @return Device
+ */
+Server::setResource('getBuildsDevice', function () {
+    return function (string $projectId) {
+        return getDevice(APP_STORAGE_BUILDS . '/app-' . $projectId);
+    };
+});
+
+/**
+ * Get cache  Device
+ * @param string $projectId of the project
+ * @return Device
+ */
+Server::setResource('getCacheDevice', function () {
+    return function (string $projectId) {
+        return getDevice(APP_STORAGE_CACHE . '/app-' . $projectId);
+    };
+});
 
 $pools = $register->get('pools');
-$connection = $pools->get('queue')->pop()->getResource();
-$workerNumber = swoole_cpu_num() * intval(App::getEnv('_APP_WORKER_PER_CORE', 6));
+$platform = new Appwrite();
+$args = $_SERVER['argv'];
 
-if (empty(App::getEnv('QUEUE'))) {
-    throw new Exception('Please configure "QUEUE" environment variable.');
+if (!isset($args[1])) {
+    Console::error('Missing worker name');
+    Console::exit(1);
 }
 
-$adapter = new Swoole($connection, $workerNumber, App::getEnv('QUEUE'));
-$server = new Server($adapter);
+\array_shift($args);
+$workerName = $args[0];
+$workerIndex = $args[1] ?? '';
```

Comment:
via the command line when the container is created , like so: php /usr/src/code/app/worker.php functions $@

Reply:
Where exactly? Because I don't see anything specified anywhere like:
 https://github.com/appwrite/appwrite/blob/20985c7ee4764d8266decae44e58068cc4168263/app/views/install/compose.phtml#L214-L241